frontend/src/Connection.ts

```diff
@@ -48,28 +62,31 @@ export class Connection {
 
   private static async trySocket(): Promise<Maybe<WebSocket>> {
     return new Promise<Maybe<WebSocket>>((resolve, _) => {
-      function clean() {
-        socket.removeEventListener('open', onSuccess);
-        socket.removeEventListener('close', onFailure);
-        socket.removeEventListener('error', onFailure);
-      }
-
-      function onSuccess() {
-        clean();
-        resolve(socket);
-      }
-
-      function onFailure() {
-        clean();
+      if (!Connection.address) {
```

Comment:
This should never be null, so the check is not necessary.

Reply:
The check actually catch the case of empty strings as well but I will remove it.